daemons/dss-checkout-sfn/app.py

```diff
@@ -0,0 +1,93 @@
+import os
+import sys
+
+import domovoi
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), 'domovoilib'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+import dss
+from dss.util.state_machine.checkout_states import definition
+from dss.util.email import send_checkout_success_email, send_checkout_failure_email
+from dss.util.checkout import parallel_copy, get_dst_bundle_prefix, get_manifest_files, \
+    validate_file_dst, validate
+
+app = domovoi.Domovoi()
+dss.Config.set_config(dss.BucketConfig.NORMAL)
+
+replica = "aws"
+
+log = dss.get_logger()
+dss_bucket = dss.Config.get_s3_bucket()
+email_sender = dss.Config.get_notification_email()
+default_checkout_bucket = dss.Config.get_s3_hca_checkout_bucket()
+
+@app.step_function_task(state_name="ScheduleCopy", state_machine_definition=definition)
+def worker(event, context):
+    bundle_id = event["bundle"]
+    version = event["version"]
+    src_bucket = dss_bucket
+    dst_bucket = get_bucket(event)
+
+    scheduled = 0
+    for src_key, dst_key in get_manifest_files(bundle_id, version, replica):
+        log.debug("Copying a file " + dst_key)
+        parallel_copy(src_bucket, src_key, dst_bucket, dst_key)
```

Comment:
Yes, since @ttung is re-writing the parallel copy we agreed to split this up. Once he is done this can be further improved by querying the state machine execution status for each copy execution